terraform/test/main.tf

```diff
@@ -40,4 +40,24 @@ module "active_directory" {
   number_of_domain_controllers = 2
   ldaps_ca_subnet              = module.networking.ldaps_ca_subnet
   environment                  = "test"
-}
\ No newline at end of file
+}
+
+resource "aws_key_pair" "bastion_ssh_key" {
+  key_name   = "bastion_ssh_key"
+  public_key = file("delta_test.pub")
```

Comment:
Currently just on my PC. Putting the private key in terraform state feels wrong, but I guess it's fine for test/staging, I'll do that.

Not necessary now, but do we have a Keeper vault set up? Might be worth doing, or getting access to the MHCLG support one

Reply:
I think we would need to get Cyber to approve of storing secrets in Keeper.

But they have approved of AWS secret manager. So we could just store secrets there?

E.g. for my marklogic setup, I'm currently considering doing this and saying that you have to manually set those secrets up:
```
data "aws_secretsmanager_secret_version" "ml_license" {
  secret_id = "ml-license-${var.environment}"
}
data "aws_secretsmanager_secret_version" "ml_admin_user" {
  secret_id = "ml-admin-user-${var.environment}"
}
```

Reply:
Sidestepped this for now by creating them in Terraform.

I think the MHCLG support team had a Keeper vault, so hopefully it would be fine.

`data` blocks still pull their values into the state though right?

Reply:
Yeah I mean my reason for doing it this way for those secrets was just so I can control the values.

If you're fine with Terraform generating a value for something, does it matter? Would anyone have access to the (decrypted) state bucket without having access to AWS secrets? We can restrict access to the bucket if needed

Reply:
Makes sense.

Probably not, I guess we just need to keep it in mind if we want to give people access to the bucket for running `terraform plan` or whatever then it has secrets in.